spring-integration-kafka/src/test/java/org/springframework/integration/kafka/dsl/KafkaDslTests.java

```diff
@@ -354,6 +371,7 @@ public DefaultKafkaHeaderMapper mapper() {
 
 			return Kafka
 					.outboundChannelAdapter(producerFactory)
+					.futuresChannel("futuresChannel")
 					.sync(true)
```

Comment:
We probably don't need `sync` to prove that "send, send, wait, wait" concept...

Reply:
It's not really important here; I just wanted to test that the futures were indeed sent to the channel; you are correct in that, in practice, you would not do both.